src/tests/system/tests/test_sssctl.py

```diff
@@ -12,274 +12,132 @@
-def test_sssctl__check_invalid_options_in_two_domains(client: Client):
+def test_sssctl__permission_and_missing_config(client: Client, setup_fn):
     """
-    :title: Verify typo in option name with multiple domains in default configuration file
+    :title: sssctl config-check detects file permission issues
     :setup:
-        1. Configure two ldap domains
-        2. Make typo in both domains
+        1. Start with valid SSSD config
+        2. Set improper permissions or remove config file
     :steps:
-        1. Call sssctl config-check
-        2. Check error message
+        1. Run sssctl config-check
     :expectedresults:
-        1. config-check detects an error
-        2. Error messages are properly set
+        1. Configuration check fails
     :customerscenario: False
-    : with other invalid option tests?
     """
-    client.sssd.common.local()
-    client.sssd.dom("ldap1")["ldap_ri"] = "ldaps://invalid"
-    client.sssd.dom("ldap1")["id_provider"] = "ldap"
-    client.sssd.dom("ldap2")["ldap_ri"] = "ldaps://invalid"
-    client.sssd.dom("ldap2")["id_provider"] = "ldap"
-    client.sssd.sssd["domains"] = "ldap1, ldap2"
-    client.sssd.config_apply(check_config=False)
-
-    res = client.sssctl.config_check()
-    assert res.rc != 0, "Config-check did not detect misconfigured config"
-    assert "Issues identified by validators: 2" in res.stdout, "Wrong number of issues found by validators"
-    assert "Attribute 'ldap_ri' is not allowed in section 'domain/ldap1'" in res.stdout, "Wrong error message"
-    assert "Attribute 'ldap_ri' is not allowed in section 'domain/ldap2'" in res.stdout, "Wrong error message"
-
-
-@pytest.mark.tools
-@pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_config_does_not_exist(client: Client):
-    """
-    :title: sssctl detects missing config
-    :setup:
-        1. Start SSSD, so default config is automatically created
-        2. Remove config
-    :steps:
-        1. Call sssctl config-check
-        2. Check error message
-    :expectedresults:
-        1. config-check detects an error
-        2. Error message is properly set
-    :customerscenario: False
-    """
-    client.sssd.common.local()
-    client.sssd.start()
-    client.fs.rm("/etc/sssd/sssd.conf")
-
+    setup_fn(client)
     result = client.sssctl.config_check()
-    assert result.rc != 0, "Config-check did not detect misconfigured config"
-    assert "File /etc/sssd/sssd.conf does not exist" in result.stdout, "Wrong error message on stdout"
+    assert result.rc != 0, "config-check should fail for file permission issue"
 
 
-@pytest.mark.tools
+@pytest.mark.importance("high")
 @pytest.mark.ticket(bz=1677994)
+@pytest.mark.tools
 @pytest.mark.topology(KnownTopology.Client)
 def test_sssctl__check_ldap_host_object_class_in_domain(client: Client):
     """
-    :title: sssctl config-check allow ldap_host_object_class in domain section
+    :title: sssctl config-check allows ldap_host_object_class in domain
```

Comment:
where ar eyou configuring ldap_host_object_class in the config?

Reply:
client.sssd.domain["ldap_host_object_class"] = "ipService"